test/dynamo/test_optimizers.py

```diff
@@ -99,6 +99,21 @@ class OptimizerTests(torch._dynamo.test_case.TestCase):
     setattr(OptimizerTests, "test_" + opt.__name__.lower(), make_test(opt))
 
 
+class MyOptimizer(torch.optim.Optimizer):
+    def __init__(self, params, val):
+        super().__init__(params, {})
+        self.val = val
+
+    def _init_group(self):
+        return self.val
+
+    def step(self):
+        ret = self._init_group()
+        # Assert that init group can return the actual value
+        assert self.val == ret
```

Comment:
maybe we can also add a test for ret-value dependent control flow? 

I think my previous attempt failed because I did not modify any tensor causing the code to not be traced

Reply:
done